thunder/dynamo/utils.py

```diff
@@ -401,6 +407,23 @@ def is_node_supported_by_thunder(node: torch.fx.Node) -> tuple[bool, SplitReason
                     return is_module_supported, split_reason
         return True, None
 
+    def get_requires_grad(arg_node):
+        if not isinstance(arg_node, torch.fx.Node):
+            return False
+
+        if "example_value" not in arg_node.meta:
+            return False
+
+        example_value = arg_node.meta["example_value"]
+        if isinstance(example_value, torch.Tensor):
+            return example_value.requires_grad
+        elif isinstance(example_value, Sequence):
+            return any(x.requires_grad for x in example_value)
+        return False
+
+    args, _ = tree_flatten((node.args, node.kwargs))
+    needs_grad = any(map(get_requires_grad, args))
```

Comment:
As this is used only in `try_execute_thunder_symbol`, I was wondering if all of this should be moved under `try_execute_thunder_symbol`.